src/libraries/System.Private.Xml/src/System/Xml/Serialization/Types.cs

```diff
@@ -605,6 +609,10 @@ internal static bool IsKnownType(Type type)
             return false;
         }
 
+        [UnconditionalSuppressMessage ("ReflectionAnalysis", "IL2118",
+            Justification = "DAM on AddPrimitive references methods of DateTime, which has a compiler-generated local function " +
+                            "LowGranularityNonCachedFallback that calls PInvokes which are considered potentially dangerous. " +
+                            "XML serialization will not access this local function.")]
```

Comment:
Maybe for future consideration - update linker to check if the called method is DataFlow interesting only because of PInvokes, and if so, run the PInvokes "safeness" check before reporting this warning.
Technically there's no reason why the PInvoke analysis needs to be part of DataFlow (it doesn't use anything from DataFlow), so decoupling it would allow us to avoid generating these warnings which are not necessary.